server/data/data-api/src/test/java/org/apache/james/rrt/lib/CanSendFromContract.java

```diff
@@ -117,7 +116,6 @@ default void allValidFromAddressesShouldContainUserAddressAndAnAliasOfTheUser()
     }
 
     @Test
-    @Disabled("Recursive aliases are not supported yet")
     default void allValidFromAddressesFromShouldBeTrueWhenSenderIsAnAliasOfAnAliasOfTheUser() throws Exception {
```

Comment:
missing at least one test: alias@domainalias -> user@domainalias -> user@domain

Reply:
it was already covered in `allValidFromAddressesShouldContainUserAddressAndAnAliasOfTheDomainUserFromAnotherDomain`

Reply:
then alias@domainalias -> alias@domain -> user@domain ?

Reply:
The case that i think we don't cover is this one : 

for a user `user@domain`.

We have a domain alias `domain2` -> `domain`

and a user alias, `alias@domain2` -> `user@domain2`.

I am not sure that we would see the address `user@domain2` as a valid header.

Reply:
Seen: does not work the other way round